### PR TITLE
Add log table widget support

### LINKED ISSUE #1831: Add support for `log-table-widget` type widgets
### Feature Description
There is no `widget_log_table` type available in the `content` for the `newrelic_one_dashboard` resource. It'd be great if there was!

### Describe Alternatives
None. I'd love to use some generic `widget_raw` kind of component that just takes widget json outright in the interim, but it doesn't seem like that exists either.

### Additional context
Add any other context here.

### PR DESCRIPTION
**Pull request by alexrosenfeld10**
# Description

Adds a `widget_log_table` resource.

~⚠️ disclaimer, this probably doesn't work. This is the first time I've ever written Go, and I'm mostly just pattern matching here. If someone who is more familiar with this wants to point me in the right direction or take this PR over, I would love it.~

Edit: this now works, thanks to @kidk 's help! 😄 

Fixes #1831 

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Checklist:

Please delete options that are not relevant.

- [ ] My commit message follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] My code is formatted to [Go standards](https://go.dev/blog/gofmt)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes. Go [here](https://github.com/newrelic/terraform-provider-newrelic#testing) for instructions on running tests locally.

## How to test this change?

Please describe how to test your changes. Include any relevant steps in the UI, HCL file(s), commands, etc

- Step 1
- Step 2
- etc

### CODE REVIEW

File: newrelic/structures_newrelic_one_dashboard.go [hunk 1]
```diff
@@ -852,7 +890,7 @@ func findDashboardWidgetFilterCurrentDashboard(d *schema.ResourceData) ([]interf
 	var widgetList []interface{}
 
 	pages := d.Get("page").([]interface{})
-	selfLinkingWidgets := []string{"widget_bar", "widget_pie", "widget_table"}
+	selfLinkingWidgets := []string{"widget_bar", "widget_pie", "widget_table", "widget_log_table"}
```
Comment (alexrosenfeld10) [inline newrelic/structures_newrelic_one_dashboard.go:893]: wasn't sure if this belongs here or not.

File: newrelic/structures_newrelic_one_dashboard.go [hunk 2]
```diff
@@ -827,6 +859,12 @@ func flattenDashboardWidget(in *entities.DashboardWidget, pageGUID string) (stri
 			out["nrql_query"] = flattenDashboardWidgetNRQLQuery(&in.Configuration.Table.NRQLQueries)
 		}
 		out["filter_current_dashboard"] = filterCurrentDashboard
+	case "logger.log-table-widget":
```
Comment (alexrosenfeld10) [inline newrelic/structures_newrelic_one_dashboard.go:862]: This is probably wrong - all the other cases here are prefixed `vis.`, and I'd imagine that's a pattern that's not meant to be broken. However, when exporting a dash that has a log table widget in it, the result looks like this:

```json
        {
          "title": "logs",
          "layout": {
            "column": 1,
            "row": 11,
            "width": 4,
            "height": 3
          },
          "linkedEntityGuids": null,
          "visualization": {
            "id": "logger.log-table-widget"
          },
          "rawConfiguration": {
            "nrqlQueries": [
              {
                "accountId": 00000,
                "query": "SELECT `log_severity`,`timestamp`,`service_name`,`level`,`message` FROM Log
              }
            ]
          }
        }
```

File: newrelic/structures_newrelic_one_dashboard.go [hunk 3]
```diff
@@ -506,6 +522,22 @@ func expandDashboardPieWidgetConfigurationInput(i map[string]interface{}, meta i
 	}
 	return nil, nil
 }
+
+func expandDashboardLogTableWidgetConfigurationInput(i map[string]interface{}, meta interface{}) (*dashboards.DashboardLogTableWidgetConfigurationInput, error) {
+	var cfg dashboards.DashboardLogTableWidgetConfigurationInput
```
Comment (alexrosenfeld10) [inline newrelic/structures_newrelic_one_dashboard.go:527]: This (and one or two other changes) require updates to `types.go` in a separate package, not sure where to go for that.

Comment (alexrosenfeld10) [reply newrelic/structures_newrelic_one_dashboard.go:527]: Ah - seems to be https://github.com/newrelic/newrelic-client-go.

File: newrelic/structures_newrelic_one_dashboard.go (hunk 1 above)
Comment (kidk) [reply newrelic/structures_newrelic_one_dashboard.go:893]: I don't think this is needed. `widget_log` doesn't seem to have filter capabilities.

File: newrelic/structures_newrelic_one_dashboard.go (hunk 2 above)
Comment (kidk) [reply newrelic/structures_newrelic_one_dashboard.go:862]: This is correct, it seems logs broke the pattern 😄

**Comment by kidk** [comment]
Thanks for the PR @alexrosenfeld10 It's awesome you took the time to add this. 

I've responded to some of your questions. We need to take a look at your last one, but we've scheduled in some time next week to do that.

File: newrelic/structures_newrelic_one_dashboard.go (hunk 2 above)
Comment (alexrosenfeld10) [reply newrelic/structures_newrelic_one_dashboard.go:758]: Thanks @kidk for your reply, but that doesn't really tell me the right way to handle the broken pattern here. Please advise

Comment (kidk) [reply newrelic/structures_newrelic_one_dashboard.go:758]: From what I can tell your code is correct, so you should be all good.

Comment (alexrosenfeld10) [reply newrelic/structures_newrelic_one_dashboard.go:758]: oh, gotcha. ok! nifty haha.

**Comment by kidk** [comment]
hi @alexrosenfeld10 

We've made some changes so it all works 🥳 

Thanks a lot for taking the time and effort to push this in. 

I'll let you do a final review, once you're happy please move the PR out of draft state.

File: newrelic/structures_newrelic_one_dashboard.go [hunk 4]
```diff
@@ -253,11 +253,11 @@ func expandDashboardPageInput(d *schema.ResourceData, pages []interface{}, meta
 				if err != nil {
 					return nil, err
 				}
-
-				widget.Configuration.LogTable, err = expandDashboardLogTableWidgetConfigurationInput(v.(map[string]interface{}), meta)
+				widget.RawConfiguration, err = expandDashboardWidgetRawConfigurationInput(v.(map[string]interface{}), meta)
 				if err != nil {
 					return nil, err
 				}
+				widget.Visualization.ID = "logger.log-table-widget"
```
Comment (alexrosenfeld10) [inline newrelic/structures_newrelic_one_dashboard.go:260]: Aha! I was wondering where this special case would finally appear, I knew it had to be somewhere.

**Comment by alexrosenfeld10** [comment]
thanks for the help getting this over the finish line @kidk , all looks good to me!

File: newrelic/resource_newrelic_one_dashboard_test.go [hunk 5]
```diff
@@ -596,7 +596,16 @@ func testAccCheckNewRelicOneDashboardConfig_PageFull(pageName string, accountID
...
+    widget_log_table {
+      title = "Log table widget"
+      row = 13
+      column = 1
+      nrql_query {
+        query      = "FROM Log SELECT *"
+      }
+    }
+
+	widget_table {
```
Comment (charryshun) [inline newrelic/resource_newrelic_one_dashboard_test.go:608]: Suggest to remove two space for format

**Pull request merged**